lib/platform-shims/deno.ts

```diff
@@ -21,22 +21,21 @@ const REQUIRE_ERROR = 'require is not supported by ESM';
 const REQUIRE_DIRECTORY_ERROR =
   'loading a directory of commands is not supported yet for ESM';
 
+const DENO_ENV_PERMITTED: boolean =
+  (await Deno.permissions.query({name: 'env'})).state === 'granted';
```

Comment:
will this work even though yargs is synchronous? (have you tested?).

Reply:
I believe that you are correct in inferring that this will cause the import mainline code to execute following other purely synchronous imports. But it will still execute before any user code. I've not seen or used any complex import recipes which would need a specific order of execution for the 'Yargs' module. 

I am successfully using a fork with this patch in several user scripts, and I can post some examples. But that's just weak, "works on my machine", evidence. What would you think would help prove that this can be used safety without causing any disruptions?

relevant: [Unexpected execution order](https://github.com/denoland/deno/issues/14243) (https://github.com/denoland/deno/issues/14243)